tests: not passing

Hello, there!

I am very excited so far about Epsilon.

First thing I tried was to run the tests, and they did not pass:

```
$ go test ./epsilon/...
--- FAIL: TestParseLocals (0.00s)
    parser_test.go:108: parsing module failed: wat2wasm failed: exit status 1
        /tmp/wat2wasm201369494/test.wat:8:7: error: unexpected token (, expected ).
                                                        (then (br 2)))
              ^
        /tmp/wat2wasm201369494/test.wat:14:4: error: unexpected token (, expected EOF.
                                (local.get $sum))
           ^
FAIL
FAIL    github.com/ziggy42/epsilon/epsilon      0.135s
FAIL
```

Just wanted to let you know. Thank you for working on this!

--- a/epsilon/simd.go
+++ b/epsilon/simd.go
@@ -2269,7 +2269,10 @@ func saturateF32toInt32(f float32) int32 {
 	if math.IsNaN(float64(f)) {
 		return 0
 	}
-	if f > math.MaxInt32 {
+
+	// We use an explicit float rather than comparing against math.MaxInt32
+	// to avoid subtle differences in behavior between architectures.
+	if f >= 2147483648.0 {
 		return math.MaxInt32
 	}
 	if f < math.MinInt32 {

